sandbox/libs/authn/docs/auth_flow.md

```diff
@@ -9,3 +9,35 @@ As requests are received by an OpenSearch node they need to be authenticated.  D
 The subject is associated for the lifetime of the request on that node.  If the request will be sent to other nodes within the cluster, the authentication information will be attached with a generated access token no matter the AuthenticationToken that was used to authenticate the user.  By including expiry on these tokens the exposure of the subjects permissions are greatly limited. 
 
 ![AccessToken for inter-node communication](https://user-images.githubusercontent.com/2754967/202580773-9b0ab15f-834c-45dc-9faf-48e6b832f85e.PNG)
+
+## Ensuring Authentication throughout OpenSearch
+
+As the OpenSearch service starts or in a running instance threads are created with the subject defaulting to Unauthenticated.  This is done to support backward compatibility as request authentication has never been a hard requirement of the service.
+
+**Why partition state by threads start?**
+Thread context is already widely used in OpenSearch to pass information through the transport layers.  This is not the most ideal state - and can be altered at a future time.  Many frameworks such as Shiro have static accessors for authentication information that is internally retrieved from a thread context.
+
+Any risks to this pattern?  Absolutely, naively usage of thread local storage for authentication information can leak into reused threads.  This is an elevation of privileges and can be hard to spot during testing, but triggerable in production scenarios.
+
+If we are authoring functionality to store Subject specific state, A) access information through stack / heap variables B) reuse known good libraries C) last resort - get a detailed review on the lifetime management of thread local storage objects
+
+### Service Startup
+As OpenSearch starts there are several management systems that trigger to discover/elect the cluster manager, monitor for node usage, snapshot, etc.  These management systems should have structured identities that ship out of the box and they are assumed directly.
+
+- [ ] Discovering all of these systems and applying least-privilege defaults is the desired end state.  This protects the system by having different layers of access that are specialized
```

Comment:
This would be great. For the most part the security plugin relies on the action name or a special header to determine if a request was issued from an internal source: https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/filter/SecurityFilter.java#L187

By creating a subject and authorizing the request using the same logic as authorizing any other request will reduce the amount of branches of logic to follow.